src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -164,38 +161,103 @@ public byte[] mask(byte[] input) {
             }
             // else, if nothing has matched, continue to the next value
             // it may happen that we don't mask anything if number / boolean masking is disabled or the value is null
-
-            maskingState.backtrackCurrentJsonPath();
         }
 
         ValueMaskingUtil.flushReplacementOperations(maskingState);
 
         return maskingState.getMessage();
     }
 
+    /**
+     * Masks the array element in place.
+     *
+     * @param maskingState the masking state an index prior to the element start index
```

Comment:
"an index prior to the element start index" => "with the index on the element prior to the element start index"??

element start index is not super clear to me either, actually.

Reply:
Actually it is only index prior the the start element index if there are no whitespaces (because those are skipped inside the method).

Reply:
I basically took the description from a similar method. But I can rename it.